src/Features/CSharp/Portable/CodeRefactorings/PushMemberUp/AbstractMemberPusher.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.  
+
+using System;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CodeActions;
+using System.Linq;
+using Microsoft.CodeAnalysis.Formatting;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.PushMemberUp
```

Comment:
why have this be in C#, there should be very little C# specific code in this feature.  We can do this at the common layer and have this work for VB as well.

This is also important because you might be pushign members *across* languages.